core/src/main/java/org/apache/brooklyn/location/byon/ByonLocationResolver.java

```diff
@@ -161,9 +161,10 @@ protected ConfigBag extractConfig(Map<?,?> locationFlags, String spec, LocationR
         String host;
         int port;
         if (ssh != null) {
-            userAndHostAndPort = parseUserAndHostAndPort((String)ssh, 22);
+            userAndHostAndPort = parseUserAndHostAndPort(ssh, 22);
         } else {
-            userAndHostAndPort = parseUserAndHostAndPort((String)winrm, 5985);
+            // TODO set to null and rely on the MachineLocation. If not then make a dependency to WinRmMachineLocation and use its config key name.
+            userAndHostAndPort = parseUserAndHostAndPort(winrm, vals.get("winrm.useHttps") != null && (Boolean)vals.get("winrm.useHttps") ? 5986 : 5985);
```

Comment:
Looks good. A more concise way of writing this is `Boolean.TRUE.equals(vals.get("winrm.useHttps"))`.